<!DOCTYPE html>
<html lang="en"><head><meta charset="UTF-8"/><meta name="viewport" content="width=device-width, initial-scale=1.0"/><title>Home · Pioneer.jl</title><meta name="title" content="Home · Pioneer.jl"/><meta property="og:title" content="Home · Pioneer.jl"/><meta property="twitter:title" content="Home · Pioneer.jl"/><meta name="description" content="Documentation for Pioneer.jl."/><meta property="og:description" content="Documentation for Pioneer.jl."/><meta property="twitter:description" content="Documentation for Pioneer.jl."/><script data-outdated-warner src="assets/warner.js"></script><link href="https://cdnjs.cloudflare.com/ajax/libs/lato-font/3.0.0/css/lato-font.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/juliamono/0.050/juliamono.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.2/css/fontawesome.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.2/css/solid.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.2/css/brands.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.16.8/katex.min.css" rel="stylesheet" type="text/css"/><script>documenterBaseURL="."</script><script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js" data-main="assets/documenter.js"></script><script src="search_index.js"></script><script src="siteinfo.js"></script><script src="../versions.js"></script><link class="docs-theme-link" rel="stylesheet" type="text/css" href="assets/themes/catppuccin-mocha.css" data-theme-name="catppuccin-mocha"/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="assets/themes/catppuccin-macchiato.css" data-theme-name="catppuccin-macchiato"/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="assets/themes/catppuccin-frappe.css" data-theme-name="catppuccin-frappe"/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="assets/themes/catppuccin-latte.css" data-theme-name="catppuccin-latte"/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="assets/themes/documenter-dark.css" data-theme-name="documenter-dark" data-theme-primary-dark/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="assets/themes/documenter-light.css" data-theme-name="documenter-light" data-theme-primary/><script src="assets/themeswap.js"></script></head><body><div id="documenter"><nav class="docs-sidebar"><div class="docs-package-name"><span class="docs-autofit"><a href>Pioneer.jl</a></span></div><button class="docs-search-query input is-rounded is-small is-clickable my-2 mx-auto py-1 px-2" id="documenter-search-query">Search docs (Ctrl + /)</button><ul class="docs-menu"><li class="is-active"><a class="tocitem" href>Home</a><ul class="internal"><li><a class="tocitem" href="#Introduction"><span>Introduction</span></a></li><li><a class="tocitem" href="#Design-Goals"><span>Design Goals</span></a></li><li><a class="tocitem" href="#Features"><span>Features</span></a></li><li><a class="tocitem" href="#Quick-Links"><span>Quick Links</span></a></li><li><a class="tocitem" href="#Authors-and-Development"><span>Authors and Development</span></a></li><li><a class="tocitem" href="#Contact"><span>Contact</span></a></li><li><a class="tocitem" href="#Exported-Methods"><span>Exported Methods</span></a></li></ul></li><li><span class="tocitem">User Guide</span><ul><li><a class="tocitem" href="user_guide/installation/">Installation Guide</a></li><li><a class="tocitem" href="user_guide/quickstart/">Quick Start Tutorial</a></li><li><a class="tocitem" href="user_guide/parameters/">Parameter Configuration</a></li></ul></li></ul><div class="docs-version-selector field has-addons"><div class="control"><span class="docs-label button is-static is-size-7">Version</span></div><div class="docs-selector control is-expanded"><div class="select is-fullwidth is-size-7"><select id="documenter-version-selector"></select></div></div></div></nav><div class="docs-main"><header class="docs-navbar"><a class="docs-sidebar-button docs-navbar-link fa-solid fa-bars is-hidden-desktop" id="documenter-sidebar-button" href="#"></a><nav class="breadcrumb"><ul class="is-hidden-mobile"><li class="is-active"><a href>Home</a></li></ul><ul class="is-hidden-tablet"><li class="is-active"><a href>Home</a></li></ul></nav><div class="docs-right"><a class="docs-navbar-link" href="https://github.com/nwamsley1/Pioneer.jl" title="View the repository on GitHub"><span class="docs-icon fa-brands"></span><span class="docs-label is-hidden-touch">GitHub</span></a><a class="docs-navbar-link" href="https://github.com/nwamsley1/Pioneer.jl/blob/develop/docs/src/index.md" title="Edit source on GitHub"><span class="docs-icon fa-solid"></span></a><a class="docs-settings-button docs-navbar-link fa-solid fa-gear" id="documenter-settings-button" href="#" title="Settings"></a><a class="docs-article-toggle-button fa-solid fa-chevron-up" id="documenter-article-toggle-button" href="javascript:;" title="Collapse all docstrings"></a></div></header><article class="content" id="documenter-page"><h2 id="Introduction"><a class="docs-heading-anchor" href="#Introduction">Introduction</a><a id="Introduction-1"></a><a class="docs-heading-anchor-permalink" href="#Introduction" title="Permalink"></a></h2><p>Pioneer and its companion tool Altimeter are an open-source and performant solution for analysis of protein MS data acquired by data-independent acquisition (DIA). Poineer includes routines for searching DIA experments from Thermo and Sciex instruments and for building spectral libraries using the <a href="https://koina.wilhelmlab.org/">Koina</a> interface. Given a spectral library of precursor fragment ion intensities and retention time estimates, Pioneer identifies and quantifies peptides from the library in the data. </p><h2 id="Design-Goals"><a class="docs-heading-anchor" href="#Design-Goals">Design Goals</a><a id="Design-Goals-1"></a><a class="docs-heading-anchor-permalink" href="#Design-Goals" title="Permalink"></a></h2><ul><li><strong>Open-Source:</strong> Pioneer is completely open source. </li><li><strong>Cross-Platform:</strong> Pioneer and the vendor-specific file conversion tool run on Linux, MacOS, and Windows</li><li><strong>High-Performance:</strong> Pioneer achieves high sensitivity, FDR control, quantitative precision and accuracy on benhcmark datat-sets </li><li><strong>Scalability:</strong> Memory consumption and speed should remain constant as the number of raw files in an anslysis grows. Pioneer should scale to very large experiments with hundreds to thousands of raw files (experimental)</li><li><strong>Fast:</strong> Pioneer searches data several times faster than it can be aquired and faster than state-of-the-art search tools.</li></ul><h2 id="Features"><a class="docs-heading-anchor" href="#Features">Features</a><a id="Features-1"></a><a class="docs-heading-anchor-permalink" href="#Features" title="Permalink"></a></h2><p>Pioneer and Altimeter build on previous search engines and introduce several new concepts:</p><ul><li><p><strong>Spectral Library Prediction with Koina</strong>: Using <a href="https://koina.wilhelmlab.org/">Koina</a> Pioneer can construct fully predicted spectral libraries given an internet connection and a FASTA file with protein sequences. Pioneer uses Chronologer to predict peptide retention times and is optimized to use Altimeter for fragment ion intensity predictions.</p></li><li><p><strong>Collision Energy Independent Spectral Libraries</strong>: Rather than predicting a single intensity value for each fragment ion, Altimeter predicts 4 B-spline coefficients. Evaluating the fragment splines at a given collision energy gives a fragment ion intensity. Pioneer calibrates the library to find the optimal collision energy value to use for each MS data file in an experiment. In this way, it is possible to use a single spectral library for different instruments and scan settings. </p></li><li><p><strong>Fragment Isotope Correction</strong>: Fragment isotope distributions depend on precursor isotope distributions as distorted by quadrupole mass filtering. Altimeter addresses this by predicting total fragment ion intensities rather than monoisotopic ones. Pioneer then accurately re-isotopes these library spectra using methods from <a href="https://pmc.ncbi.nlm.nih.gov/articles/PMC6166224/">Goldfarb et al.</a>. This is particularly important for narrow-window DIA methods where precursor isotopic envelopes frequently straddle multiple windows.</p></li><li><p><strong>Qaudrupole Transmission Modeling</strong>: For narrow-window data, Pioneer can optionally estimate a quadrupole-transmission efficiency function for more accurate re-isotoping. </p></li><li><p><strong>Intensity-Aware Fragment Index Search</strong>: Pioneer implements a fast fragment index search inspired by <a href="https://pubmed.ncbi.nlm.nih.gov/28394336/">MSFragger</a> and <a href="https://pubmed.ncbi.nlm.nih.gov/37819886/">Sage</a>. Pioneer&#39;s implementation uniquely leverages accurate fragment intensity predictions from in silico libraries to improve both speed and specificity of the search.</p></li><li><p><strong>Linear Regression onto Library Templates</strong>: Pioneer explains each observed mass spectrum as a linear combination of template spectra from the library. To reduce quantitative bias from interfering signals, Pioneer minimizes the <a href="https://en.wikipedia.org/wiki/Huber_loss">pseudo-Huber</a> loss rather than squared error. This provides robust quantification even in complex spectra. For other examples of linear regression applied to DIA analyses, see <a href="https://pubmed.ncbi.nlm.nih.gov/29608554/">Specter</a> and <a href="https://www.biorxiv.org/content/10.1101/2024.05.27.596040v2">Chimerys</a>.</p></li><li><p><strong>Scalability</strong>: Pioneer was designed to scale to large experiments with many MS data files. Memory consumption remains constant as the number of data files in an experiment grows large. </p></li></ul><h2 id="Quick-Links"><a class="docs-heading-anchor" href="#Quick-Links">Quick Links</a><a id="Quick-Links-1"></a><a class="docs-heading-anchor-permalink" href="#Quick-Links" title="Permalink"></a></h2><ul><li><a href="user_guide/installation/#Installation-Guide">Installation Guide</a></li><li><a href="user_guide/quickstart/#Quick-Start-Tutorial">Quick Start Tutorial</a></li></ul><h2 id="Authors-and-Development"><a class="docs-heading-anchor" href="#Authors-and-Development">Authors and Development</a><a id="Authors-and-Development-1"></a><a class="docs-heading-anchor-permalink" href="#Authors-and-Development" title="Permalink"></a></h2><p>Pioneer is developed and maintained by:</p><ul><li>Nathan Wamsley (<a href="https://majorlab.wustl.edu/">Major Lab</a>/<a href="https://goldfarblab.wustl.edu/">Goldfarb Lab</a>, Washington University)</li><li>Dennis Goldfarb (<a href="https://goldfarblab.wustl.edu/">Goldfarb Lab</a>, Washington University)</li></ul><h2 id="Contact"><a class="docs-heading-anchor" href="#Contact">Contact</a><a id="Contact-1"></a><a class="docs-heading-anchor-permalink" href="#Contact" title="Permalink"></a></h2><p>For questions about Pioneer or to collaborate, please contact:</p><ul><li>Nathan Wamsley (wamsleynathan@gmail.com)</li><li>Dennis Goldfarb (dennis.goldfarb@wustl.edu)</li></ul><p>For toubleshooting use the <a href="https://github.com/nwamsley1/Pioneer.jl/issues">Issues page</a> on github. To critique methods or propose features use the <a href="https://github.com/nwamsley1/Pioneer.jl/discussions">Discussions page</a>.</p><h2 id="Exported-Methods"><a class="docs-heading-anchor" href="#Exported-Methods">Exported Methods</a><a id="Exported-Methods-1"></a><a class="docs-heading-anchor-permalink" href="#Exported-Methods" title="Permalink"></a></h2><ul><li><a href="#Pioneer.BuildSpecLib"><code>Pioneer.BuildSpecLib</code></a></li><li><a href="#Pioneer.GetBuildLibParams"><code>Pioneer.GetBuildLibParams</code></a></li><li><a href="#Pioneer.GetSearchParams"><code>Pioneer.GetSearchParams</code></a></li><li><a href="#Pioneer.SearchDIA"><code>Pioneer.SearchDIA</code></a></li><li><a href="#Pioneer.convertMzML"><code>Pioneer.convertMzML</code></a></li></ul><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="Pioneer.SearchDIA" href="#Pioneer.SearchDIA"><code>Pioneer.SearchDIA</code></a> — <span class="docstring-category">Function</span><span class="is-flex-grow-1 docstring-article-toggle-button" title="Collapse docstring"></span></header><section><div><pre><code class="language-julia hljs">SearchDIA(params_path::String)</code></pre><p>Main entry point for the DIA (Data-Independent Acquisition) search workflow. Executes a series of <code>SearchMethod</code>s and generates performance metrics.</p><p>Parameters:</p><ul><li>params_path: Path to JSON configuration file containing search parameters</li></ul><p>Output:</p><ul><li>Generates a log file in the results directory</li><li>Long and wide-formatted tables (.arrow and .csv) for protein-group and precursor level id&#39;s and quantitation.</li><li>Reports timing and memory usage statistics</li></ul><p>Example:</p><pre><code class="language-julia hljs">julia&gt; SearchDIA(&quot;/path/to/config.json&quot;)
==========================================================================================
Sarting SearchDIA
==========================================================================================

Starting search at: 2024-12-30T14:01:01.510
Output directory: ./../data/ecoli_test/ecoli_test_results
[ Info: Loading Parameters...
[ Info: Loading Spectral Library...
 .
 .
 .</code></pre><p>If it does not already exist, SearchDIA creates the user-specified results_dir and generates quality control plots, data tables, and logs.</p><pre><code class="nohighlight hljs">results_dir/
├── pioneer_search_log.txt
├── qc_plots/
│   ├── collision_energy_alignment/
│   │   └── nce_alignment_plots.pdf
│   ├── quad_transmission_model/
│   │   ├── quad_data
│   │   │   └── quad_data_plots.pdf
│   │   └── quad_models
│   │       └── quad_model_plots.pdf
│   ├── rt_alignment_plots/
│   │   └── rt_alignment_plots.pdf
│   ├── mass_error_plots/
│   │   └── mass_error_plots.pdf
│   └── QC_PLOTS.pdf
├── precursors_long.arrow
├── precursors_long.tsv
├── precursors_wide.arrow
├── precurosrs_wide.tsv
├── protein_groups_long.arrow
├── protein_groups_long.tsv
├── protein_groups_wide.arrow
└── protein_groups_wide.tsv</code></pre></div><a class="docs-sourcelink" target="_blank" href="https://github.com/nwamsley1/Pioneer.jl/blob/3e97721b89bc914ebc54285569ff598b8936995f/src/Routines/SearchDIA.jl#L67-L122">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="Pioneer.GetSearchParams" href="#Pioneer.GetSearchParams"><code>Pioneer.GetSearchParams</code></a> — <span class="docstring-category">Function</span><span class="is-flex-grow-1 docstring-article-toggle-button" title="Collapse docstring"></span></header><section><div><pre><code class="language-julia hljs">GetSearchParams(lib_path::String, ms_data_path::String, results_path::String; 
               params_path::Union{String, Missing} = missing,
               simplified::Bool = true)</code></pre><p>Creates a search parameter configuration file with user-specified paths.</p><p>The function loads default parameters from either the simplified or full JSON template (from assets/example_config/) and creates a customized parameter file with the user&#39;s file paths. All other parameters retain their default values and can be modified later.</p><p>Arguments:</p><ul><li>lib_path: Path to the spectral library file (.poin)</li><li>ms<em>data</em>path: Path to the MS data directory  </li><li>results_path: Path where search results will be stored</li><li>params<em>path: Output path for the parameter file. Can be a directory (creates search</em>parameters.json)  or full file path. Defaults to &quot;search_parameters.json&quot; in current directory.</li><li>simplified: If true (default), uses simplified template with essential parameters only.  If false, uses full template with all advanced options.</li></ul><p>Returns:</p><ul><li>String: Path to the newly created search parameters file</li></ul><p>Templates used:</p><ul><li>Simplified: <code>defaultSearchParamsSimplified.json</code> (basic parameters)</li><li>Full: <code>defaultSearchParams.json</code> (all advanced parameters)</li></ul><p>Example:</p><pre><code class="language-julia hljs"># Create simplified parameter file
output_path = GetSearchParams(
    &quot;/path/to/speclib.poin&quot;,
    &quot;/path/to/ms/data/dir&quot;, 
    &quot;/path/to/results/dir&quot;
)

# Create full parameter file with custom output location
output_path = GetSearchParams(
    &quot;/path/to/speclib.poin&quot;,
    &quot;/path/to/ms/data/dir&quot;,
    &quot;/path/to/results/dir&quot;;
    params_path = &quot;/custom/path/my_params.json&quot;,
    simplified = false
)</code></pre></div><a class="docs-sourcelink" target="_blank" href="https://github.com/nwamsley1/Pioneer.jl/blob/3e97721b89bc914ebc54285569ff598b8936995f/src/Routines/GenerateParams.jl#L132-L177">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="Pioneer.BuildSpecLib" href="#Pioneer.BuildSpecLib"><code>Pioneer.BuildSpecLib</code></a> — <span class="docstring-category">Function</span><span class="is-flex-grow-1 docstring-article-toggle-button" title="Collapse docstring"></span></header><section><div><pre><code class="language-julia hljs">BuildSpecLib(params_path::String)</code></pre><p>Main function to build a spectral library from parameters. Executes a series of steps:</p><ol><li>Parameter validation and directory setup</li><li>Fragment bound detection</li><li>Retention time prediction (optional)</li><li>Fragment prediction (optional)</li><li>Library index building</li></ol><p>Parameters:</p><ul><li>params_path: Path to JSON configuration file containing library building parameters</li></ul><p>Output:</p><ul><li>Generates a spectral library in the specified output directory</li><li>Creates a detailed log file with timing and performance metrics</li><li>Returns nothing</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/nwamsley1/Pioneer.jl/blob/3e97721b89bc914ebc54285569ff598b8936995f/src/Routines/BuildSpecLib.jl#L40-L57">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="Pioneer.GetBuildLibParams" href="#Pioneer.GetBuildLibParams"><code>Pioneer.GetBuildLibParams</code></a> — <span class="docstring-category">Function</span><span class="is-flex-grow-1 docstring-article-toggle-button" title="Collapse docstring"></span></header><section><div><pre><code class="language-julia hljs">GetBuildLibParams(out_dir::String, lib_name::String, fasta_inputs; 
                 params_path::Union{String, Missing} = missing,
                 regex_codes::Union{Missing, Dict, Vector} = missing,
                 simplified::Bool = true)</code></pre><p>Creates a library building parameter configuration file with user-specified paths and FASTA files.</p><p>The function loads default parameters from either the simplified or full JSON template  (from assets/example_config/) and creates a customized parameter file with the user&#39;s paths and automatically discovered FASTA files. All other parameters retain their default values and can be modified later.</p><p>Arguments:</p><ul><li>out_dir: Output directory path where the library will be built</li><li>lib_name: Name for the spectral library (used for directory and file naming)</li><li>fasta_inputs: FASTA file specification. Can be:<ul><li>A single directory path (String) - searches for .fasta/.fasta.gz files</li><li>A single FASTA file path (String) </li><li>An array of directories and/or FASTA file paths</li></ul></li><li>params<em>path: Output path for the parameter file. Can be a directory (creates buildspeclib</em>params.json) or full file path. Defaults to &quot;buildspeclib_params.json&quot; in current directory.</li><li>regex_codes: Optional FASTA header regex patterns for protein annotation extraction. Can be:<ul><li>A single Dict with keys: &quot;accessions&quot;, &quot;genes&quot;, &quot;proteins&quot;, &quot;organisms&quot; (applied to all FASTA files)</li><li>A Vector of Dicts for positional mapping to fasta_inputs</li><li>If missing, uses default patterns from the template</li></ul></li><li>simplified: If true (default), uses simplified template with essential parameters only. If false, uses full template with all advanced library building options.</li></ul><p>Returns:</p><ul><li>String: Path to the newly created library building parameters file</li></ul><p>Templates used:</p><ul><li>Simplified: <code>defaultBuildLibParamsSimplified.json</code> (basic parameters)</li><li>Full: <code>defaultBuildLibParams.json</code> (all advanced parameters)</li></ul><p>The function automatically:</p><ul><li>Discovers FASTA files in specified directories</li><li>Generates appropriate library names from FASTA filenames</li><li>Expands regex patterns to match the number of FASTA files found</li><li>Validates that all specified paths exist and are accessible</li></ul><p>Example:</p><pre><code class="language-julia hljs"># Create simplified parameter file with directory of FASTA files
output_path = GetBuildLibParams(
    &quot;/path/to/output&quot;, 
    &quot;my_library&quot;,
    &quot;/path/to/fasta/directory&quot;
)

# Create full parameter file with specific FASTA files and custom regex
output_path = GetBuildLibParams(
    &quot;/path/to/output&quot;,
    &quot;my_library&quot;, 
    [&quot;/path/to/human.fasta&quot;, &quot;/path/to/yeast.fasta&quot;];
    params_path = &quot;/custom/path/build_params.json&quot;,
    regex_codes = Dict(&quot;accessions&quot; =&gt; &quot;^sp\|(\w+)\|&quot;, &quot;genes&quot; =&gt; &quot; GN=(\S+)&quot;),
    simplified = false
)</code></pre></div><a class="docs-sourcelink" target="_blank" href="https://github.com/nwamsley1/Pioneer.jl/blob/3e97721b89bc914ebc54285569ff598b8936995f/src/Routines/GenerateParams.jl#L225-L286">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="Pioneer.convertMzML" href="#Pioneer.convertMzML"><code>Pioneer.convertMzML</code></a> — <span class="docstring-category">Function</span><span class="is-flex-grow-1 docstring-article-toggle-button" title="Collapse docstring"></span></header><section><div><pre><code class="language-julia hljs">convertMzML(mzml_dir::String; skip_scan_header::Bool=true)</code></pre><p>Convert mzML mass spectrometry data files to Arrow IPC format.</p><p>Takes either a directory containing mzML files or a path to a single mzML file and converts them to  Arrow format, preserving scan data including m/z arrays, intensity arrays, and scan metadata.</p><p><strong>Arguments</strong></p><ul><li><code>mzml_dir::String</code>: Path to either a directory containing mzML files or a path to a single mzML file</li><li><code>skip_scan_header::Bool=true</code>: When true, omits scan header information from the output to reduce file size</li></ul><p><strong>Returns</strong></p><p><code>nothing</code></p><p><strong>Output</strong></p><p>Creates Arrow (.arrow) files in the same directory as the input mzML files and with the same base filename.</p><p><strong>Examples</strong></p><pre><code class="language-julia hljs"># Convert all mzML files in a directory
convertMzML(&quot;path/to/mzml/files&quot;)

# Convert a single mzML file
convertMzML(&quot;path/to/single/file.mzML&quot;)

# Include scan headers in output
convertMzML(&quot;path/to/mzml/files&quot;, skip_scan_header=false)</code></pre><p><strong>Notes</strong></p><p>Each mzML file is converted to a corresponding Arrow IPC (.arrow) file in the same directory. This is particularly useful for Sciex data where direct .wiff/.wiff2 conversion is not supported</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/nwamsley1/Pioneer.jl/blob/3e97721b89bc914ebc54285569ff598b8936995f/src/Routines/mzmlConverter/convertMzML.jl#L296-L328">source</a></section></article></article><nav class="docs-footer"><a class="docs-footer-nextpage" href="user_guide/installation/">Installation Guide »</a><div class="flexbox-break"></div><p class="footer-message">Powered by <a href="https://github.com/JuliaDocs/Documenter.jl">Documenter.jl</a> and the <a href="https://julialang.org/">Julia Programming Language</a>.</p></nav></div><div class="modal" id="documenter-settings"><div class="modal-background"></div><div class="modal-card"><header class="modal-card-head"><p class="modal-card-title">Settings</p><button class="delete"></button></header><section class="modal-card-body"><p><label class="label">Theme</label><div class="select"><select id="documenter-themepicker"><option value="auto">Automatic (OS)</option><option value="documenter-light">documenter-light</option><option value="documenter-dark">documenter-dark</option><option value="catppuccin-latte">catppuccin-latte</option><option value="catppuccin-frappe">catppuccin-frappe</option><option value="catppuccin-macchiato">catppuccin-macchiato</option><option value="catppuccin-mocha">catppuccin-mocha</option></select></div></p><hr/><p>This document was generated with <a href="https://github.com/JuliaDocs/Documenter.jl">Documenter.jl</a> version 1.14.1 on <span class="colophon-date" title="Friday 10 October 2025 20:13">Friday 10 October 2025</span>. Using Julia version 1.11.7.</p></section><footer class="modal-card-foot"></footer></div></div></div></body></html>
